robonomics/io/src/source/virt.rs

```diff
@@ -59,6 +61,22 @@ pub fn pubsub(
     Ok(pubsub.subscribe(&topic_name).map(|v| Ok(v)))
 }
 
+/// Read data records from blockchain.
+///
+/// Returns datalog data objects.
+pub fn datalog(
+    remote: String,
+) -> Result<(
+    impl Sink<AccountId, Error = Error>,
+    impl Stream<Item = Result<Vec<RingBufferItem>>>,
```

Comment:
Generally, we don’t need internal information about pallet structure outside of this, let’s return primitive data instead of RingBufferItem.